src/content/1.7/development/components/grid-component.md

```diff
@@ -7,229 +7,404 @@ weight: 2
+    
+        $productGridFactory = $this->get('prestashop.core.grid.product_grid_factory');
+        $productGrid = $productGridFactory->getGrid($searchCriteria);
         
-        return $this->render('@Foo/Bar/pageWithGrid.html.twig', [
-            'grid' => $gridPresenter->present($grid),
+        return $this->render('@PrestaShop/Admin/Product/products.html.twig', [
+            // $this->presentGrid() is helper method provided by FrameworkBundleAdminController
+            'productsGrid' => $this->presentGrid($productGrid),
         ]);
     }
 }
 ```
 
-And in the related template:
+To see Grid in your page you have to include it's template which is provided by PrestaShop.
 
 ```twig
-    {{ include('@PrestaShop/Admin/Common/Grid/grid_panel.html.twig', {'grid': grid }) }}
-```
-
-## Grid component rendering using Twig templating engine 
-
-This component is built with 2 main principles in mind:
-
-* the data is never aware of his representation, this means we rely on Presenters to present the data to the view;
-* we can alter every part of the Grid component, but everything is frozen once it's build;
-
-### Managing the data and the UI
-
-This means the Grid Data doesn't contains anything like *width* or *class* or *color* because we give the responibility of Twig to allow such customization of this kind.
-
-> Even if the component is powerful and flexible enough, you shouldn't try to configure Grid to pass this kind of data related to the view.
-
-In the other hand, the view if totaly customizable: you can override every available template for a column, a column header or a column filter. You can also override a template for a specific grid or globally, you can change the view only according to special business conditions too.
-
-The right and only way to change the data is using hooks and module overrides won't works here while the UI can be changed using templating and Javascript.
-
-To sum up this strict separation between the Data management and his UI representation, this is how you could imagine the creation of a Grid:
-
-{{< figure src="../img/grid_workflow.png" title="Grid Workflow" >}}
-
-> Note a XML file for this schema is importable in services like [draw.io](https://draw.io) is [available](/schemas/1.7/grid_workflow.xml).
+{# @PrestaShop/Admin/Product/products.html.twig #}
 
+{% include '@PrestaShop/Admin/Common/Grid/grid_panel.html.twig' with {'grid': productsGrid} %}
+```
 
-### Once built, everything is frozen
+It is possible to include provided template and modify some parts of it or you can create your own template to render
+Grid. 
 
-You will notice that almost every object build is *final* and have no setters: we avoid every chance to break the Grid component once the build of the Grid have been done and validated.
+## Learn more
 
-But we got you covered, you can alter almost everything *before* the Grid is presented to the view thanks to the available hooks:
+* [How to work with Bulk actions](#)
```

Comment:
If you add these links, you need to complete them :angel:

Reply:
can I add _Comming soon_ and make different PR's for that?